src/libraries/System.Private.CoreLib/src/System/Threading/ThreadLocal.cs

```diff
@@ -175,6 +177,8 @@ protected virtual void Dispose(bool disposing)
             lock (s_idManager)
             {
                 id = ~_idComplement;
+                int slotId = id / valuePerSlot;
+                int withinSlotId = id % valuePerSlot;
```

Comment:
`Math.DivRem`? 🤔

Reply:
Note that `valuePerSlot` is a power-of-two constant. Do all JIT optimizations for power-of-two constants still kick in when this DivRem call?